examples/post-sync/main.go

```diff
@@ -0,0 +1,322 @@
+				fmt.Sprintf("%s: %s", conditionType, condition.Message))
+			if commit == "" && condition.Commit != "" {
+				commit = condition.Commit
+			}
+		}
+	}
+
+	return strings.Join(errorMessages, "\n"), commit
+}
+
+func (c *SyncStatusController) processRepoSyncConditions(conditions []v1beta1.RepoSyncCondition) (string, string) {
+	var errorMessages []string
+	var commit string
+
+	for _, condition := range conditions {
+		isErrorCondition := false
+		var conditionType v1beta1.RepoSyncConditionType
+
+		switch condition.Type {
+		case v1beta1.RepoSyncStalled:
+			if condition.Status == metav1.ConditionTrue {
+				isErrorCondition = true
+				conditionType = condition.Type
+			}
+		case v1beta1.RepoSyncReconcilerFinalizerFailure:
+			if condition.Status == metav1.ConditionTrue {
+				isErrorCondition = true
+				conditionType = condition.Type
+			}
+			// Other condition types (Reconciling, Syncing, ReconcilerFinalizing)
+			// are not treated as primary error indicators here for aggregation.
+		}
+
+		if isErrorCondition {
+			errorMessages = append(errorMessages,
+				fmt.Sprintf("%s: %s", conditionType, condition.Message))
+			if commit == "" && condition.Commit != "" {
+				commit = condition.Commit
+			}
+		}
+	}
+
+	return strings.Join(errorMessages, "\n"), commit
+}
+
+// isSuccessfulSync checks if the rsync status no errors in rsync status and all commits match.
+func isSuccessfulSync(status v1beta1.Status) bool {
+	allCommitsMatch := status.Source.Commit != "" &&
+		status.Source.Commit == status.Rendering.Commit &&
+		status.Rendering.Commit == status.Sync.Commit
+
+	return allCommitsMatch && !hasStatusError(status)
+}
+
+// hasStatusError checks if there are any errors in the rsync status.
+func hasStatusError(status v1beta1.Status) bool {
+	return status.Source.Errors != nil ||
+		status.Rendering.Errors != nil ||
+		status.Sync.Errors != nil
+}
+
+// extractErrorAndCommitFromStatus extracts errors and the associated commit from the status.
+func extractErrorAndCommitFromStatus(status v1beta1.Status) (string, string) {
+	if status.Source.Errors != nil {
+		return aggregateErrors(status.Source.Errors), status.Source.Commit
+	}
+	if status.Rendering.Errors != nil {
+		return aggregateErrors(status.Rendering.Errors), status.Rendering.Commit
+	}
+	return aggregateErrors(status.Sync.Errors), status.Sync.Commit
+}
+
+// aggregateErrors aggregates error messages from a slice of ConfigSyncError
+// and returns a single error message.
+func aggregateErrors(errors []v1beta1.ConfigSyncError) string {
+	if len(errors) == 0 {
+		return ""
+	}
+
+	var errorMessages []string
+	for _, err := range errors {
+		errorMessages = append(errorMessages, fmt.Sprintf("Code: %s, Message: %s", err.Code, err.ErrorMessage))
+	}
+
+	return "aggregated errors: " + strings.Join(errorMessages, "\n")
+}
+
+// Update the StatusTracker to use SyncID
+type StatusTracker struct {
+	mu   sync.Mutex
+	seen map[SyncID]map[string]struct{}
```

Comment:
Implement an eviction strategy (TTL, LRU, max size) to prevent unbounded memory growth over time. This can be done in a follow-up PR as an improvement.

Reply:
Ack